In [2]:
import numpy as np
import pandas as pd
from scipy import stats

In [3]:
df = pd.read_csv('marketing_AB.csv')

In [4]:
df.head()

Unnamed: 0  user id test group  converted  total ads most ads day  \
0           0  1069124         ad      False        130       Monday   
1           1  1119715         ad      False         93      Tuesday   
2           2  1144181         ad      False         21      Tuesday   
3           3  1435133         ad      False        355      Tuesday   
4           4  1015700         ad      False        276       Friday   

   most ads hour  
0             20  
1             22  
2             18  
3             10  
4             14

In [5]:
df.describe()

Unnamed: 0       user id      total ads  most ads hour
count  588101.000000  5.881010e+05  588101.000000  588101.000000
mean   294050.000000  1.310692e+06      24.820876      14.469061
std    169770.279667  2.022260e+05      43.715181       4.834634
min         0.000000  9.000000e+05       1.000000       0.000000
25%    147025.000000  1.143190e+06       4.000000      11.000000
50%    294050.000000  1.313725e+06      13.000000      14.000000
75%    441075.000000  1.484088e+06      27.000000      18.000000
max    588100.000000  1.654483e+06    2065.000000      23.000000

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 588101 entries, 0 to 588100
Data columns (total 7 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   Unnamed: 0     588101 non-null  int64 
 1   user id        588101 non-null  int64 
 2   test group     588101 non-null  object
 3   converted      588101 non-null  bool  
 4   total ads      588101 non-null  int64 
 5   most ads day   588101 non-null  object
 6   most ads hour  588101 non-null  int64 
dtypes: bool(1), int64(4), object(2)
memory usage: 27.5+ MB


In [13]:
df['test group'].value_counts()

ad     564577
psa     23524
Name: test group, dtype: int64

In [14]:
df['converted'].value_counts()

False    573258
True      14843
Name: converted, dtype: int64

In [16]:
contingency_table = pd.crosstab(df['test group'], df['converted'] , margins = True)

In [17]:
contingency_table

converted    False   True     All
test group                       
ad          550154  14423  564577
psa          23104    420   23524
All         573258  14843  588101

In [18]:
d = np.array([contingency_table.iloc[0][: -1].values,contingency_table.iloc[1][: -1].values])

In [19]:
d

array([[550154,  14423],
       [ 23104,    420]], dtype=int64)

In [20]:
stats.chi2_contingency(d)

(54.005823883685245,
 1.9989623063390075e-13,
 1,
 array([[550327.71899045,  14249.28100955],
        [ 22930.28100955,    593.71899045]]))

## Chi-square Test of Independence Results

- Chi-square test statistic: 54.006
- Degrees of freedom: 1
- p-value: 1.999e-13 (approximately)

The chi-square test of independence was performed to assess the relationship between the advertising campaign (ad vs. psa) and the conversion of users (purchase or not) in the online store.

The observed data was analyzed using a contingency table, which includes two categories: the ad category (users who saw advertisements) and the psa category (users who did not see advertisements). The conversion of users (purchase or not) was recorded in the "converted" column.

Based on the results of the chi-square test, the chi-square test statistic was calculated to be approximately 54.006. The degrees of freedom associated with the test were 1. The extremely small p-value, approximately 1.999e-13, indicates strong evidence against the null hypothesis of independence between the advertising campaign and user conversion.

This finding suggests a significant association between the advertising campaign and the conversion of users in the online store. The expected frequencies, calculated assuming independence between the variables, differ substantially from the observed frequencies.

Therefore, we can conclude that the advertising campaign has a statistically significant impact on user conversion in the online store. Users who saw the advertisements (ad category) exhibited different conversion patterns compared to those who did not see the advertisements (psa category).

---

Please note that this report is based on the interpretation of the chi-square test results and provides a general understanding of the relationship between the advertising campaign and user conversion. You may need to customize the report further based on specific details and context related to your online store and data analysis.